lib/remap-functions/src/decode_base64.rs

```diff
@@ -0,0 +1,88 @@
+use remap::prelude::*;
+
+#[derive(Clone, Copy, Debug)]
+pub struct DecodeBase64;
+
+impl Function for DecodeBase64 {
+    fn identifier(&self) -> &'static str {
+        "decode_base64"
+    }
+
+    fn parameters(&self) -> &'static [Parameter] {
+        &[Parameter {
+            keyword: "value",
+            accepts: |v| matches!(v, Value::Bytes(_)),
+            required: true,
+        }]
+    }
+
+    fn compile(&self, mut arguments: ArgumentList) -> Result<Box<dyn Expression>> {
+        let value = arguments.required("value")?.boxed();
+
+        Ok(Box::new(DecodeBase64Fn { value }))
+    }
+}
+
+#[derive(Clone, Debug)]
+struct DecodeBase64Fn {
+    value: Box<dyn Expression>,
+}
+
+impl Expression for DecodeBase64Fn {
+    fn execute(&self, state: &mut state::Program, object: &mut dyn Object) -> Result<Value> {
+        let value = self.value.execute(state, object)?.try_bytes()?;
+
+        match base64::decode(value) {
+            Ok(v) => Ok(Value::from(v)),
+            Err(_) => Err("unable to decode value to base64".into()),
+        }
+    }
+
+    fn type_def(&self, state: &state::Compiler) -> TypeDef {
+        self.value
+            .type_def(state)
+            .fallible_unless(value::Kind::Bytes)
```

Comment:
```suggestion
            .into_fallible(true)
```

Based on the documentation in https://docs.rs/base64/0.13.0/base64/enum.DecodeError.html, I believe this function will have to be marked as _always fallible_, since it depends on the input bytes if decoding succeeds or not.